HerPortal.BusinessLogic/ExternalServices/S3FileReader/S3FileReader.cs

```diff
@@ -97,4 +97,16 @@ public async Task<IEnumerable<S3Object>> GetS3ObjectsByCustodianCodeAsync(string
             throw;
         }
     }
+
+    public async Task<bool> FileExistsAsync(string custodianCode, int year, int month)
+    {
+        var key = keyService.GetS3KeyFromData(custodianCode, year, month);
+        var request = new ListObjectsV2Request
+        {
+            BucketName = config.BucketName,
+            Prefix = key,
+        };
+        var files = await s3Client.ListObjectsV2Async(request);
```

Comment:
I don't like using a List call fo this.

We shouldn't rely on the prefix only matching the whole key, and there is a suggestion on this StackOverflow comment that the AWS account will be charged for a List call: https://stackoverflow.com/questions/8303011/how-to-check-if-a-specified-key-exists-in-a-given-s3-bucket-using-java/36653034#comment26415871_16589633

From other answers on that question, it looks like we should be using some sort of `GetObjectMetadataAsync()` call with appropriate exception handling is the only way to reliably achieve the desired outcome here. It's not nice, but it looks to be the only approach that we have.

Reply:
made this change - i suppose on the above prefix comment it in theory should be fine (key is the entire file name including .csv), though it's not immediately clear this is so

Reply:
While it will almost certainly be fine in practice, there's nothing to stop a file from being called `referrals.csv__backup` or similar.

While we don't expect that to happen, it's better to be precise.